react-app/components/Gallery/index.jsx

```diff
@@ -1,42 +1,89 @@
-import { useContext, memo } from 'react';
+import { memo, useEffect, useState } from 'react';
 import ImageGallery from 'react-image-gallery';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faXmark } from '@fortawesome/free-solid-svg-icons';
 import 'react-image-gallery/styles/css/image-gallery.css';
-import { GalleryContext } from '@contexts/galleryContext';
+import globalObject from '@constants/globalObject';
+import apiConfig from '@config/api';
 
 import './styles.css';
-// TODO: Remove this images ulr's when files come from backend
-const images = [
-  {
-    original: 'https://fastly.picsum.photos/id/1018/1000/600.jpg?hmac=8y6PgnvgTLEEW-118lVn6V6zPUVSN9JSi27GSpmGpdQ',
-    thumbnail: 'https://fastly.picsum.photos/id/1018/250/150.jpg?hmac=a9jeH9XHKHtbPkhvklrYWgcEOX9kfjewcCNGPOrbAs0'
-  },
-  {
-    original: 'https://picsum.photos/id/1015/1000/600/',
-    thumbnail: 'https://picsum.photos/id/1015/250/150/'
-  },
-  {
-    original: 'https://picsum.photos/id/1019/1000/600/',
-    thumbnail: 'https://picsum.photos/id/1019/250/150/'
-  }
-];
+
+const itemsPerPage = 10;
 
 function Gallery() {
-  const { isGalleryOpened, setIsGalleryOpened } = useContext(GalleryContext);
-  // TODO: uncomment this when files come from backend
-  // const formatListImagesGallery = filesToUploadList.map(({ url}) => ({ original: url, thumbnail: url }));
-  const toggleGallery = () => setIsGalleryOpened(!isGalleryOpened);
+  const [imagesList, setImagesList] = useState([]);
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [sizeItems, setSizeItems] = useState(0);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [fetchedPages, setFetchedPages] = useState({});
+  const [isLoading, setIsLoading] = useState(false);
+  const currentPageLastItem = currentPage * itemsPerPage - 1;
+
+  const handleOnThumbnailClick = (_, index) => {
+    setCurrentIndex(index);
+  };
+
+  const handleOnSlide = (currentIndex) => {
+    setCurrentIndex(currentIndex);
+  };
+
+  const fetchUploadedFiles = async (page = 0) => {
```

Comment:
what about using this type of in-line documentation? https://github.com/openedx/frontend-app-course-authoring/blob/6f238bdbe0bfe19da6a8fef5b9f05f0e9b30bb46/src/pages-and-resources/data/api.js#L26-L31

Reply:
Just in case:

https://jsdoc.app/

Reply:
I do not like jsdoc, it says which types need to be passed but someone can send another type but I will use it in this case